Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2390,12 +2442,14 @@ extension BrowserViewController: ContextMenuHelperDelegate {
 }
 
 extension BrowserViewController: KeyboardHelperDelegate {
-
     func keyboardHelper(keyboardHelper: KeyboardHelper, keyboardWillShowWithState state: KeyboardState) {
         keyboardState = state
-        // if we are already showing snack bars, adjust them so they sit above the keyboard
-        if snackBars.subviews.count > 0 {
-            adjustFooterSize(nil)
+        updateViewConstraints()
+
+        UIView.animateWithDuration(state.animationDuration) {
+            UIView.setAnimationCurve(state.animationCurve)
+            self.findInPageContainer.layoutIfNeeded()
+            self.snackBars.layoutIfNeeded()
```

Comment:
I think what you're looking for here is `setNeedsLayout`. Also its usually good practice to call `view.layoutIfNeeded` before the animation call to flush out any pending animations.